### DISCUSSION TITLE
incomplete backup, process stops with error referring to SMS

### DISCUSSION THREAD

**Discussion by obacht8** [mrrfv/open-android-backup#63] [Q&A] [answered]
Hi, 
Cool project! Heard about it in the /e/OS-forum, happy to have an option for backing up my phone.

Yet still something goes wrong for me when I try to backup my Samsung S7 (SM-G930F) running /e/OS 1.11Q (which is based on LOS): the backup process stops with an error that relates to something going wrong with backup of SMS (?)

I tried “tar” and “adb”-variant of backup, the outcome was the same: the backup ended with an error because something did not work out with SMS-backup (missing folder or something like that ?). The result was a tmp-backup-folder (inside the package-folder) containing folders for 
•	contacts (VCF - but CSV was expected) 
•	apps (apk-files – they do not contain user specific data when restored, right?)
•	SMS (empty folder...)
In followup tries the content of the tmp-folder was deleted and new identical content was written but then same error: no SMS-backup.

Not sure if because of the before mentioned error but the backup was not written to where it should have been written (according to previous folder selection in the process). 
I assume the “SMS-error” caused the backup to stop at this point and to not proceed with: 
•	Photos and videos in their original quality.
•	Internal storage - includes Signal backups, downloads, and other important data.
•	Call logs, exported into a text file

Any ideas how this can be solved?

Feedback on UI:
Not sure if that is OS -specific but it is not sufficient to only enable “developer options” but one must enable “ADB debugging” in developer options… only then the connection is being made. From the displayed information that was not clear to me … but I found out after tweaking and changing around different cables and USB-ports. So I suggest to be more specific in the UI/description and name it: “enable ADB debugging” (not sure if that´s correct English wording as my system is set to German language)

Question: there was no interaction with the app needed as it got installed and deinstalled without asking questions…– is that expected behaviour? (and if Yes: What is the “export”-button for?)

best regards!

**Comment by mrrfv** [comment, marked as answer]
> Question: there was no interaction with the app needed as it got installed and deinstalled without asking questions…– is that expected behaviour?

I think is is the problem.

Open Android Backup uses the app to export data that's hard to obtain through `adb`. **When the script opens the app on your device, you must manually press the "Export data" button.** If you forget to press the button, the script won't know what to do as the files it expects are missing.

I might better document this in the future because sometimes I forget to press the button too, especially during testing (working on new features at the moment).

Could you please try again, remembering to interact with the app?

**Reply by obacht8** [reply]
will do this evening and report here (thanks for your quick answer)

**Reply by obacht8** [reply]
> When the script opens the app on your device, you must manually press the "Export data" button

Yes, that did the trick!
(I must admit that in the first place I oversaw that it is even clearly advised in the UI ...)

**Reply by adsbenham** [reply]
I've just started using Open Android Backup (thanks for developing it).  My Android 10 tablet backup was simple, only needed the 'Export data' button.  On my Android 11 smartphone two more steps were needed - to answer a question requesting that it could access all files, and then to grant Android permissions for the app to make phone calls and access call logs.  Without those options enabled, the backup would fail.
Ah, just realised the Android tablet doesn't have phone capabilities, so that would explain not needing permissions!
